Source/KakapoDB.swift

```diff
@@ -17,47 +17,95 @@ enum KakapoDBError: ErrorType {
     case InvalidId
 }
 
+private final class ArrayBox<T> {
+    private init(_ value: [T]) {
+        self.value = value
+    }
+    
+    private var value: [T]
+}
+
 class KakapoDB {
     
-    private var _uuid = 0
-    var store: [String: [KStorable]] = [:]
+    private let queue = dispatch_queue_create("com.kakapodb.queue", DISPATCH_QUEUE_CONCURRENT)
+    private var _uuid = -1
+    private var store: [String: ArrayBox<KStorable>] = [:]
     
-    func create<T: KStorable>(_: T.Type, number: Int) {
-        var array = lookup(T)
-        for _ in 0..<number {
-            array.append(T(id: uuid()))
+    func create<T: KStorable>(_: T.Type, number: Int) -> [KStorable] {
+        var result: [KStorable] = []
+        
+        dispatch_barrier_sync(queue) { [weak self] in
+            guard let weakSelf = self else { return }
+            
+            let arrayBox = weakSelf.lookup(T)
+            result = (0..<number).map { _ in T(id: weakSelf.uuid()) }
+            arrayBox.value.appendContentsOf(result)
+            weakSelf.store[String(T)] = arrayBox
```

Comment:
No need to get it and set it back to the dictionary (that also mutate it),  right?